### ISSUE TITLE
K-truss benchmarking

### ISSUE THREAD

**Issue by ChuckHastings** [rapidsai/cugraph#4593]
Benchmark k-truss and compare to other existing implementations.

**Comment by jnke2016** [comment]
In 24.06, we updated our K-Truss implementation by moving from cuHornet to an implementation matching our primitive and the performance improvement was several order of magnitude faster. 

As a summary, our earlier [implementation](https://github.com/rapidsai/cugraph/pull/4438) run several iteration of our primitive `per_v_pair_dst_nbr_intersection` to compute the number of triangles incident to every edge in the graph. This can be inefficient if there is only a few subset of the edges to invalidate which will still require to run `per_v_pair_dst_nbr_intersection` on the entire edgelist and that primitive is not well optimized for K-Truss. 
In our new implementation, we do not compute the number of triangles incident to every edge but instead identify weak edges to be deleted and remove their contribution on the affected triangles. This optimization improves our performance for different graph size and k values as it can be seeing in the figure below

<img width="749" alt="Image" src="https://github.com/user-attachments/assets/817ec237-1d46-4beb-a1fb-89eb1c286d43" />

**Comment by ChuckHastings** [comment]
This is a great start.  But I think we should also show comparison to other implementations (I saw nx, hornet, arkuda in a quick internet search, but any we can find).

**Comment by jeaton32** [comment]
These are all single-gpu so far? Which GPU is being used?

**Comment by jnke2016** [comment]
> These are all single-gpu so far

Yes I do also have single node multi GPU numbers which will be added shortly.

>  Which GPU is being used

NVIDIA H100 80GB HBM3

**Comment by jnke2016** [comment]
Runtime breakdown on a single GPU for a graph generated at scale 22 and edge factor 16.

<img width="1402" alt="Image" src="https://github.com/user-attachments/assets/22fb2baf-ab4e-4dcb-a41d-21ac83136d53" />
